src/aspire/utils/BlockDiagonal.pyx

```diff
@@ -0,0 +1,561 @@
+import numpy as np
+
+from numpy.linalg import norm
+from numpy.linalg import solve
+from scipy.linalg import block_diag
+from scipy.special import jv
+
+from aspire.utils.cell import Cell2D
+from aspire.basis.ffb_2d import FFBBasis2D
+from aspire.basis.basis_utils import lgwt
+
+from aspire.utils.blk_diag_func import blk_diag_apply
+from aspire.utils.blk_diag_func import blk_diag_solve
+
+SCALAR_TYPES=(int, float, complex)
```

Comment:
Probably we should use the standard definition from Numpy instead of redefining here.

Reply:
How do you mean? I've never seen.  Please advise.  (Numpy dtype will compare true with python types using this if that is what you were concerned about)

Reply:
upon review, numpy dtype objects are curiously complicated.  I made an internal function and it just returns np.isscalar for now. That should do for now.